src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -0,0 +1,372 @@
+        if obs_start <= source_vol.get_observation_value(obs) <= obs_end
+    ]
+    observations = observations[::obs_stride]
+
+    for obs in observations:
+        extents = source_vol.get_extents(obs)
+        bases = source_vol.get_bases(obs)
+        quadratures = source_vol.get_quadratures(obs)
+        tensor_names = source_vol.list_tensor_components(obs)
+        grid_names = source_vol.get_grid_names(obs)
+
+        if components_to_interpolate is not None:
+            tensor_names = list(
+                set(tensor_name for pattern in components_to_interpolate
+                    for tensor_name in tensor_names
+                    if re.match(pattern, tensor_name)))
+
+        # pre-load all tensors to avoid loading the full tensor for each element
+        tensors = [
+            np.array(source_vol.get_tensor_component(obs, name), copy=False)
+            for name in tensor_names
+        ]
+
+        volume_data = []
+        index = 0  # keeps track of position in tensor components
+        # iterate over elements
+        for i, extent in enumerate(extents):
+            source_mesh = Mesh[dim](extent, basis_from_string(bases[i]),
+                                    quadrature_from_string(quadratures[i]))
+
+            interpolant = RegularGrid[dim](source_mesh, target_mesh)
+            grid_points = source_mesh.number_of_grid_points()
+
+            tensor_comps = []
+            # iterate over tensors
+            for j, tensor in enumerate(tensors):
+                dataVector = DataVector(tensor[index:index + grid_points],
+                                        copy=False)
+                interpolated_tensor = interpolant.interpolate(dataVector)
+                tensor_path = os.path.join(grid_names[i], tensor_names[j])
+                tensor_comps.append(
+                    TensorComponent(
+                        tensor_path, DataVector(interpolated_tensor,
+                                                copy=False)))
+
+            volume_data.append(
+                ElementVolumeData(target_mesh.extents(), tensor_comps,
+                                  target_mesh.basis(),
+                                  target_mesh.quadrature()))
+            index += grid_points
+
+        assert index == len(tensors[0])
+        target_vol.write_volume_data(obs,
+                                     source_vol.get_observation_value(obs),
+                                     volume_data)
+
+    source_file.close()
+    target_file.close()
+
+
+def _parse_args(sys_args):
+    """
+    defines and parses the command line arguments
+
+    Parameters
+    ----------
+    sys_args arguments passed via command line
+
+    Returns
+    -------
+    A dictionary of the parsed command line arguments
+    """
+    parser = argparse.ArgumentParser(
+        description=
+        "Interpolate the tensor components of h5 files to a desired grid")
+
+    parser.add_argument("source_files",
```

Comment:
I think instead of the file list / glob expansion here you can just name this argument `--file-prefix` (with `required=True`) and load all files with the prefix followed by a number, as is done in `GenerateXdmf`. I like the way you handle the file input here, but I think it's better to be consistent with the other scripts. Maybe we can convert the three scripts together to do the glob file input later.

Reply:
you mean like `type = lambda prefix: glob.glob(prefix + "[0-9]*.h5")` . I understand that consistency is important but the original solution seems very simple to understand for anyone whos ever used a terminal in this case. Your call.

Reply:
If you make the option `--source-file-prefix` then, in addition to consistency with `GenerateXdmf, it also simplifies the `--target-file-prefix` option because you can reliably do the 1-1 mapping. So I would suggest to not evaluate the glob right here in the option, but use it to build the target file list as well.